### ISSUE TITLE
Setting times on PV display?

### ISSUE THREAD

**Issue by knyckis** [rooklift/ogatak#121]
As far as I can tell, currently there is no way of setting the "speed" of the PV function, in the sense that that the pace of the move played out when you hover over the position can be changed. In Katrain, the similar function plays out the moves much slower, making it far easier to get a sense of the flow. In Ogatak, the move sequences are virtually immediate, making it much harder to follow. In addition, it often fluctuates (when the engine changes its mind about the best sequence, I presume), making it even harder.

My suggestion, if feasible, is to allow the user to select the 'move pace' of the PV function, so that it plays out the entire sequence (the length of which I can already select, thanks for that) on that pace, and only then starts over if the engine thinks there is an even better one.

Thanks for a great program - I really prefer the look of this compared to e.g. Katrain.

**Comment by rooklift** [comment]
While it's possible in principle, it's not so trivial to do...

**Comment by knyckis** [comment]
I see. Perhaps the standard setting could be "slowed" then, more similar to katrain. Apart from the technical viability, that of course depends on whether it would be preferable. I do think, though, that for most people, it is cognitively easier to follow a sequence if it is not (virtually) immediate. Getting a flow of the moves is important, I think, and currently that is somewhat hard, at least for a slow thinker like myself :)

**Comment by rooklift** [comment]
> Perhaps the standard setting could be "slowed" then

That's the part that's not trivial to do... currently we are just doing the whole thing instantly.

**Comment by rooklift** [comment]
Well, roughly what has to happen is;

* When the mouse goes over a move, the PV is stored.
* The draw function is repeatedly called on the desired schedule, to advance the PV display.
* But this cycle is aborted if the mouse moves away, or the node changes, or we enter editing mode, etc.

**Comment by knyckis** [comment]
This obviously comes from a place of technical ignorance, but it seems like it is the second step where you possibly could modify what is displayed. For example, since the full set of moves now are displayed pretty much instantly, could you not store the full set and then display it with a selected pace, and only then call on a new set with whatever is the new node at that time (if the user still holds the mouse on the same spot)? Then it seems (again, speaking in pure fantasy/thought experiment terms) as if you would get an uninterrupted sequence that the user can digest, followed by new ones for as long as the user wants them.

Of course, I do not really expect to help solve this problem by wishful thinking, so I will let you ponder of the viability of the enterprise in peace now.

Thanks for your super quick replies – you are following the same speedy information-giving pattern as your program it seems :)

**Comment by rooklift** [comment]
Yeah it's just a bit fiddly to code I think. Really Ogatak is the GUI that I want, and if other people use it that's great, but I might not spend too much effort on features I won't use myself. :)

**Comment by knyckis** [comment]
Gotcha. Makes complete sense, and I for one is really grateful for it. I use it on mac and have to go through a little process when I want to update version, but it is definitely worth it. I find the GUI so much aesthetically pleasing than e.g. Katrain. My suggestion here is pretty much the only thing I find more useful in Katrain than in Ogatak. Being able to see what the engine "has in mind" as continuation is really fantastic. I just find that so many moves instantly, and the constant fluctuations, makes me use it less than I would otherwise.

Again, thanks for putting it out there for us to freely use!

**Comment by MianmianYingxiang** [comment]
> Yeah it's just a bit fiddly to code I think. Really Ogatak is the GUI that I want, and if other people use it that's great, but I might not spend too much effort on features I won't use myself. :)
> 
> btw when I said "what has to happen is" - I meant this is what I would have to add.

I think Ogatak just needs to be optimized and there is no need for other less relevant functions.
